### PR TITLE
[libc++] Fix ambiguity when using std::scoped_allocator constructor

### LINKED ISSUE #78754: [libc++] scoped_allocator_adaptor(Convertible, Convertible) shouldn't be ambiguous
Snippet:

```C++
#include <memory_resource>
#include <vector>
#include <string>
#include <cassert>
#include <scoped_allocator>

using PA = std::pmr::polymorphic_allocator<std::pmr::string>;

int main() {
    std::pmr::monotonic_buffer_resource mr1;
    std::pmr::monotonic_buffer_resource mr2;

    using A = std::scoped_allocator_adaptor<PA, PA>;
    A scoped_alloc = A(&mr1, &mr2);

    std::vector<std::pmr::string, PA> v;
    v.emplace_back("foo");
    assert(v.get_allocator().resource() == &mr1);
    return (v[0].get_allocator().resource() == &mr2);
}
```

Output:

```C++
error: call to constructor of 'A' (aka 'scoped_allocator_adaptor<polymorphic_allocator<basic_string<char, std::char_traits<char>, polymorphic_allocator<char>>>, polymorphic_allocator<basic_string<char, std::char_traits<char>, polymorphic_allocator<char>>>>') is ambiguous
```

[GodBolt](https://godbolt.org/z/1Ev8ch3Wx)

### PR DESCRIPTION
**Pull request by ldionne**
As a drive-by change, also fix a name that wasn't uglified in the code and add a test for that.

Fixes #78754

### CODE REVIEW

File: libcxx/include/scoped_allocator [hunk 1]
```diff
@@ -197,6 +197,8 @@ struct __get_is_always_equal<_A0, _Allocs...> {
   static const bool value = allocator_traits<_A0>::is_always_equal::value && __get_is_always_equal<_Allocs...>::value;
 };
 
+struct __chained_ctor_tag {};
```
Comment (mordante) [inline libcxx/include/scoped_allocator:200]: Can you add some comment to this tag, preferably referring to the bug it fixes.

**Review by mordante — APPROVED** [review]
LGTM modulo one comment.
I spend a bit of time hunting for the real changes, I would have preferred to split this in two commits.

**Comment by ldionne** [comment]
@mordante No worries, I've split up the drive-by change into https://github.com/llvm/llvm-project/pull/80706.

**Comment by ldionne** [comment]
I'm abandoning this in favour of #80261

**Pull request closed without merging**